app/src/androidTest/java/com/android/streetworkapp/ui/parkoverview/ParkRatingTest.kt

```diff
@@ -108,4 +112,60 @@ class ParkRatingTest {
       assert(starRating.intValue == index)
     }
   }
+
+  // New tests after MVVM link :
+
+  // 1 - Is the button correctly hidden / shown depending on user and park state :
+
+  @Test
+  fun isRatingButtonCorrectlyDisplayedWhenNoRating() {
+    val showRatingDialog = mutableStateOf(false)
+    // default park constructor
+    val emptyPark =
+        Park("", "", ParkLocation(0.0, 0.0, ""), "", 1f, 1, 10, 0, emptyList(), emptyList())
+
+    val user = User("uid", "username", "email", 0, emptyList())
+
+    composeTestRule.setContent { ParkDetails(emptyPark, showRatingDialog, user) }
+
+    // User not null and has not rated the emptyPark (empty votersUID list).
+    composeTestRule.onNodeWithTag("ratingButton").assertIsDisplayed()
+  }
+
+  @Test
+  fun isRatingButtonCorrectlyDisplayedWhenRating() {
+    val showRatingDialog = mutableStateOf(false)
+    val user = User("uid", "username", "email", 0, emptyList())
+
+    // default park constructor
+    val emptyPark =
```

Comment:
you used showRatingDialog and emptypark twice, so initiating it in a @before `setup function` would be a good practice

Reply:
Empty park can be put inside the setup function